### DISCUSSION TITLE
Expand Doesn't work as expected

### DISCUSSION THREAD

**Discussion by paragopinath** [Shopify/function-examples#260] [Feedback: Cart Transform API]
In Cart, we have 3 bundles of products and out of that only one bundle product display the expanded view in the checkout while the rest of the 2 products doesn't expand even though the function operations resulted in expands

function log: 
<img width="464" alt="Screen Shot 2023-06-13 at 10 22 30 AM" src="https://github.com/Shopify/function-examples/assets/56590166/61f1ac77-57aa-40ae-b22e-edbde0925c24">

checkout: 
<img width="497" alt="Screen Shot 2023-06-13 at 10 22 47 AM" src="https://github.com/Shopify/function-examples/assets/56590166/f9346596-3b5b-4612-b60c-c625d3846f0c">

Function Log ID: `/01H2DH8M3XFAYFN01SE9DFWP69/function_runs/5afaac85-2ae4-4774-a4ac-e88f0606584c`

**Comment by paragopinath** [comment]
Update: I've tested this in another store with checkout extensibility enabled and I'm still seeing the same issue

Function log:
<img width="827" alt="Screen Shot 2023-06-14 at 10 11 56 AM" src="https://github.com/Shopify/function-examples/assets/56590166/45f1b0cf-e6e2-4795-b769-2207626d71ee">

Checkout:
<img width="687" alt="Screen Shot 2023-06-14 at 10 12 12 AM" src="https://github.com/Shopify/function-examples/assets/56590166/8380b241-3d02-4fbf-a6e5-abce77a99a9a">

**Comment by ignacio-chiazzo** [comment]
Hi! 

We currently don't support having all quantities greater than 1. At least one of the components must have quantity 1. 

We will remove this restriction **very soon**, so both operations will apply. At the same time, we're working on making logs better so that it's easier to answer the question _"why didn't my operation apply?"_

**Comment by paragopinath** [comment]
Thanks for clarifying that @ignacio-chiazzo!
I've tested with quantity 1 and it worked!

I have another question unrelated to this before we close this thread, should the meta fields definitions (namespace, key) be the same as documented [here](https://shopify.dev/docs/apps/selling-strategies/bundles/create#create-metafields-on-product-variants) or can we use custom definitions to save the configuration as long as we write the function in a way it results in expand/merge operations we should be okay? Does Shopify rely on these meta fields for anything downstream after an order is placed? 

Thanks for your time!

**Reply by ignacio-chiazzo** [reply]
Glad to hear that, and again, we will remove that restriction soon, so those operations will apply. 

> Does Shopify rely on these meta fields for anything downstream after an order is placed?

No, Shopify does not rely on any reserved namespace. It's up to the app to use the name they need. 

Thanks for the feedback!

**Reply by BrandoCommando** [reply]
Personally, I find the way Shopify suggests (component parents/reference/quantities) pretty painful. It seems it would only useful when there are very few permutations of parents/children. That's why we've implemented a custom Cart Transform function (using JS no less!) with totally different metafields and logic. It all works fine once you get the hang of it! Now, if they would just make the Cart Transform function transform the cart (and not just checkout), and we'll be in business!

**Reply by ignacio-chiazzo** [reply]
> It seems it would only useful when there are very few permutations of parents/children.

We are looking for feedback on the API. I have seen you proposed some changes (we're already on it), please keep sending us feedback.

> now, if they would just make the Cart Transform function transform the cart (and not just checkout),

We do currently transform the cart, it isn't only checkout. If your store isn't getting the cart transformed, then it's likely a bug in our side (or a configuration of the test shop).
